src/client/Microsoft.Identity.Client/Internal/Requests/IntegratedWindowsAuthRequest.cs

```diff
@@ -38,9 +38,22 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
             await ResolveAuthorityAsync().ConfigureAwait(false);
             await UpdateUsernameAsync().ConfigureAwait(false);
             var userAssertion = await FetchAssertionFromWsTrustAsync().ConfigureAwait(false);
-            var msalTokenResponse = await SendTokenRequestAsync(
-                                            GetAdditionalBodyParameters(userAssertion), cancellationToken)
-                                        .ConfigureAwait(false);
+
+            MsalTokenResponse msalTokenResponse;
+            try
+            {
+                msalTokenResponse = await SendTokenRequestAsync(
+                                                GetAdditionalBodyParameters(userAssertion), cancellationToken)
+                                            .ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                throw new MsalClientException(
+                    MsalError.IntegratedWindowsAuthenticaitonFailed,
+                    "There was an error during integrated windows authenticaiton. This may occur if there is an issue with your ADFS configuration."
```

Comment:
```suggestion
                    "Failure to acquire a token with integrated windows authentication. This may occur if there is an issue with your ADFS configuration."
```